### PR TITLE
Add cell validation [#170897317]

### PR DESCRIPTION
**Pull request by dougmartin**
Ready for review after feedback.

### CODE REVIEW

File: src/shared/components/data-table-field.tsx [hunk 1]
```diff
@@ -261,6 +268,43 @@ export const DataTableField: React.FC<FieldProps> = props => {
...
+    const propType = fieldDefinition[propName].type;
+    if ((propType === "number") || (propType === "integer")) {
+      const {minimum, maximum} = fieldDefinition[propName] as IDataTableNumberInputField;
+      let numericValue: number;
+      if (propType === "integer") {
+        numericValue = parseInt(value, 10);
+        if (isNaN(numericValue)) {
+          return {valid: false, error: "Not a whole number"};
+        }
+        if (!value.trim().match(/^[-+]?[0-9]*$/)) {
```
Comment (knowuh) [inline src/shared/components/data-table-field.tsx]: Naive me thinks this regex test might be redundant. 

It must be to catch edge cases where the value us the input includes something parseable plus extra junk like `d4 xsdfsdf wxxx`  -- it might parse to `4` but the input still should be rejected because its clearly nonsense.

I guess I answered my own question.  Seems good.

File: src/shared/index.tsx [hunk 2]
```diff
@@ -286,3 +286,79 @@ ReactDOM.render(
...
+                "type": "integer",
+                "minimum": -100,
+                "maximum": 100
+              },
+              "validation": {
+                "title": "Float (-100 to 100)",
+                "type": "number",
+                "minimum": -100,
+                "maximum": 100
+              }
```
Comment (knowuh) [inline src/shared/index.tsx]: My question here is:  What causes one of these items to use validation, and one not to? Is it the label `noValidation` or ?

File: src/shared/components/data-table-field.tsx (hunk 1 above)
Comment (dougmartin) [reply src/shared/components/data-table-field.tsx]: It also catches this: `parseInt("20.", 10) === 20`

File: src/shared/index.tsx (hunk 2 above)
Comment (dougmartin) [reply src/shared/index.tsx]: I changed it when I added integer validation and forgot to update the label. :)
